ofonotest/qml/ofonotest/main.qml

```diff
@@ -1,74 +1,148 @@
-    }
-    Text {
-        id: textLine2
-        anchors.top: textLine.bottom
-        text: manager.available ? netreg.name : "Ofono not available"
+
+    Column {
+
+        anchors.fill: parent
+
+        Text {
+            text: context1.active ? "online" : "offline"
+        }
+
+//        Text {
+//            text:  "modem Manufacturer " + modem1.manufacturer
+//        }
+
+        Text {
+            id: textLine2
+
+            text: manager.available ? netreg.name : "Ofono not available"
+        }
+
+        Text {
+            text: "simManager.pinRequired " + simManager.pinRequired
+            visible: simManager.pinRequired !== OfonoSimManager.NoPin
+        }
+
+        Repeater {
+            model: manager.modems
+
+            Text {
+                text: modelData
+                font.bold: modelData === manager.defaultModem
+            }
+
+        }
+
     }
+
     MouseArea {
         anchors.fill: parent
         onClicked: {
-          context1.active = !context1.active
+            context1.active = !context1.active;
+            console.log("context1.active " + context1.active);
         }
     }
 
     OfonoManager {
         id: manager
+
         onAvailableChanged: {
-            console.log("Ofono is " + available)
-           textLine2.text = manager.available ? netreg.currentOperator["Name"].toString() :"Ofono not available"
+            console.log("Ofono is " + available);
+            console.log("netreg.currentOperatorPath: " + netreg.currentOperatorPath);
+            textLine2.text = (manager.available && netreg.currentOperator !== undefined) ? netreg.currentOperator["Name"].toString() : "Ofono not available";
         }
         onModemAdded: {
-            console.log("modem added "+modem)
+            console.log("modem added " + modem);
+        }
+        onModemRemoved: {
+            console.log("modem removed");
+        }
+        onModemsChanged: {
+            console.log("ModemsChanged " + manager.modems);
         }
-        onModemRemoved: console.log("modem removed")
     }
 
     OfonoConnMan {
-       id: ofono1
-       Component.onCompleted: {
-       //    console.log(manager.modems)
-       }
-       modemPath: manager.modems[0]
+        id: ofono1
+
+        modemPath: manager.defaultModem
+        onValidChanged: {
+            if (ofono1.valid) {
+                console.log("OfonoConnMan.contexts " + ofono1.contexts);
+            }
+        }
     }
 
     OfonoModem {
         id: modem1
-       modemPath: manager.modems[0]
 
+        modemPath: manager.defaultModem
+
+        onValidChanged: {
+            console.log(modem1.modemPath + ": modem1.valid " + modem1.valid);
+            if (modem1.valid) {
+//                console.log("modem1.name " + modem1.name)
+//                console.log("modem1.model " + modem1.model)
+                console.log("modem1.manufacturer " + modem1.manufacturer);
+                console.log("modem1.features " + modem1.features);
```

Comment:
These could have colon. Now I'm getting output of just "modem1.manufacturer" which is a bit strange.

Reply:
I prefer to use semicolons explicitly because of javascript have some evil cases when it joins multiple lines into one expression. If I understand correctly, you have some guideline to avoid them if possible, so I removed them.